h/views/admin/users.py

```diff
@@ -97,9 +96,7 @@ def users_rename(request):
 
     try:
         svc = request.find_service(name="rename_user")
-        svc.check(user, new_username)
-
-        rename_user.delay(user.id, new_username)
+        svc.rename(user, new_username)
```

Comment:
The view was calling `RenameUserService.check()` first so that it can catch any errors and report them back to the admin user. Then it was scheduling a Celery task (`rename_user.delay()`) that would call `RenameUserService.rename()` in a Celery worker and `rename()` would [call `check()` again](https://github.com/hypothesis/h/blob/c5a260d97e4d47fc879161ec5656ca179a69d17b/h/services/rename_user.py#L43).

The view now calls `RenameUserService.rename()` directly (instead of scheduling a Celery task to call it) so `check()` gets called within the web request without the view needing to call it itself. (`RenameUserService.rename()` is now fast since it just enqueues jobs instead of actually reindexing annotations itself, so the view can call it directly).